From 8a6c60908e9b4e40bcd7e262d5ab980987f31e2e Mon Sep 17 00:00:00 2001
From: Jan Pavlinec <jan.pavlinec@nic.cz>
Date: Tue, 6 Mar 2018 15:21:46 +0100
Subject: [PATCH] libarchive: update to version 3.3.2 (security fix)

* CVE-2017-13813 - found by OSS-Fuzz
* CVE-2017-13816 - found by OSS-Fuzz
* CVE-2017-13812 - found by OSS-Fuzz

Issue #https://gitlab.labs.nic.cz/turris/turris-os-packages/issues/171
---
 libs/libarchive/Makefile | 52 +++++++++++++++++++++++++++++++++---------------
 1 file changed, 36 insertions(+), 16 deletions(-)

diff --git a/libs/libarchive/Makefile b/libs/libarchive/Makefile
index 9c42aad..090af68 100644
--- a/libs/libarchive/Makefile
+++ b/libs/libarchive/Makefile
@@ -8,12 +8,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libarchive
-PKG_VERSION:=3.1.2
+PKG_VERSION:=3.3.2
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://www.libarchive.org/downloads
-PKG_MD5SUM:=efad5a503f66329bb9d2f4308b5de98a
+PKG_HASH:=ed2dbd6954792b2c054ccf8ec4b330a54b85904a80cef477a1c74643ddafa0ce
 PKG_MAINTAINER:=Johannes Morgenroth <morgenroth@ibr.cs.tu-bs.de>
 PKG_LICENSE:=BSD-2-Clause
 
@@ -22,37 +22,50 @@ PKG_FIXUP:=autoreconf
 
 include $(INCLUDE_DIR)/package.mk
 
-define Package/$(PKG_NAME)/Default
+define Package/libarchive/Default
 	SECTION:=libs
 	CATEGORY:=Libraries
-	DEPENDS:=+zlib
+	DEPENDS:=+zlib +liblzma +libbz2 +libexpat
 	TITLE:=Multi-format archive and compression library
 	URL:=http://www.libarchive.org/
 endef
 
-define Package/$(PKG_NAME)
-	$(call Package/$(PKG_NAME)/Default)
+define Package/libarchive
+	$(call Package/libarchive/Default)
 	DEPENDS += +libopenssl
 endef
 
-define Package/$(PKG_NAME)-noopenssl
-	$(call Package/$(PKG_NAME)/Default)
+define Package/libarchive-noopenssl
+	$(call Package/libarchive/Default)
 	TITLE += (without OpenSSL dependency)
 	VARIANT:=noopenssl
 endef
 
+define Package/bsdtar
+	SECTION:=utils
+	CATEGORY:=Utilities
+	SUBMENU:=Compression
+	DEPENDS:=+libarchive-noopenssl
+	TITLE:=BSD variant that supports various file compression formats
+	URL:=http://www.libarchive.org/
+endef
+
+define Package/bsdtar/description
+	Reads a variety of formats including tar, pax, zip, xar, lha, ar,
+	cab, mtree, rar, warc, 7z and ISO images. Writes tar, pax, zip,
+	xar, ar, ISO, mtree and shar archives. Automatically handles
+	archives compressed with gzip, bzip2, lzip, xz, lzma or compress.
+endef
+
 CONFIGURE_ARGS += \
 	--disable-bsdcpio \
-	--disable-bsdtar \
+	--enable-bsdtar=shared \
 	--disable-acl \
 	--disable-xattr \
-	--without-bz2lib \
-	--without-lzma \
-	--without-lzmadec \
-	--without-lzo2 \
 	--without-nettle \
-	--without-expat \
-	--without-xml2
+	--without-xml2 \
+	--without-lz4 \
+	--without-cng \
 
 ifeq ($(BUILD_VARIANT),noopenssl)
 	CONFIGURE_ARGS += --without-openssl
@@ -71,7 +84,14 @@ define Package/libarchive/install
 	$(INSTALL_DIR) $(1)/usr/lib
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/*.so.* $(1)/usr/lib/
 endef
-Package/$(PKG_NAME)-noopenssl/install = $(Package/$(PKG_NAME)/install)
+
+define Package/bsdtar/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(CP) $(PKG_INSTALL_DIR)/usr/bin/bsdtar $(1)/usr/bin
+endef
+
+Package/libarchive-noopenssl/install = $(Package/libarchive/install)
 
 $(eval $(call BuildPackage,libarchive))
 $(eval $(call BuildPackage,libarchive-noopenssl))
+$(eval $(call BuildPackage,bsdtar))
-- 
2.7.4

